src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1191,23 +1204,27 @@ private CSharpSyntaxNode GetBindingRootOrInitializer(CSharpSyntaxNode node)
             return bindingRoot;
         }
 
-        private IOperation GetStatementOrRootOperation(CSharpSyntaxNode node, CancellationToken cancellationToken)
+#nullable enable
+        private IOperation GetRootOperation()
         {
-            Debug.Assert(node == GetBindingRootOrInitializer(node));
+            BoundNode highestBoundNode = GetBoundRoot();
+            Debug.Assert(highestBoundNode != null);
 
-            BoundNode highestBoundNode;
-            GetBoundNodes(node, out _, out _, out highestBoundNode, out _);
-
-            // decide whether we should use highest or lowest bound node here 
-            // https://github.com/dotnet/roslyn/issues/22179
-            BoundNode result = highestBoundNode;
+            if (highestBoundNode is BoundGlobalStatementInitializer { Statement: var innerStatement })
+            {
+                // We don't have an IOperation node for the top-level statement here.
```

Comment:
>top-level statement [](start = 60, length = 19)

When I read this, I think about regular top-level statements and immediately feel confused because I think that we cannot get here for them. It looks like this is very script specific and is not for a regular statement. Consider clarifying this.